bread/layout/components/search.py

```diff
@@ -1,69 +1,112 @@
+import typing
+
 import htmlgenerator as hg
 from django.utils.translation import gettext_lazy as _
 
 from .icon import Icon
 from .loading import Loading
 
 
+class SearchBackendConfig(typing.NamedTuple):
+    url: typing.Any
+    result_selector: str = None
+    result_label_selector: str = None
+    result_value_selector: str = None
+    query_parameter: str = "q"
+
+
 class Search(hg.DIV):
-    def __init__(self, size="xl", placeholder=None, widgetattributes=None, **kwargs):
+    def __init__(
+        self,
+        size="xl",
+        placeholder=None,
+        widgetattributes=None,
+        backend=None,
+        resultcontainerid=None,
+        show_result_container=True,
+        resultcontainer_onload_js=None,
+        disabled=False,
+        **kwargs,
+    ):
+        """
+        :param SearchBackendConfig backend: Where and how to get search results
+        """
         kwargs["_class"] = kwargs.get("_class", "") + f" bx--search bx--search--{size}"
         kwargs["data_search"] = True
         kwargs["role"] = "search"
 
-        attributes = {
+        widgetattributes = {
             "id": "search__" + hg.html_id(self),
             "_class": "bx--search-input",
             "type": "text",
             "placeholder": placeholder or _("Search"),
             **(widgetattributes or {}),
         }
+        if backend:
+            resultcontainerid = (
+                resultcontainerid or f"search-result-{hg.html_id((self, backend.url))}"
```

Comment:
Can we make sure that ```resultcontainerid``` is handled without the use of ```html_id```? It leads to duplicate html ids if we copy the component inside an inline-formset. I would do it the same way as in search_select.py: ```resultcontainerid = f"search-result-attributes['id']}"```

Reply:
Remark: In the case that the Search-component is used standalone it is no problem to use html_id, but if we are inside a FormSet it will be (same is true for SearchSelect).

Reply:
Remark: Why it works inside FormSet forms with the widget-id is that Django also generates a widget ID for the template form (the one used for adding new forms) but the ID uses ```__prefix__``` instead of an integer number (as for all other places inside the empty-form-template where the number of the form would occur) which is then later replaced by the (client-side) increased value of the form counter.

Reply:
![screenshot-1625114339_246x454](https://user-images.githubusercontent.com/3115209/124065700-800dc100-da61-11eb-98fa-fa04d192f49f.png)
Just to clarify the actual problem: The two inputs (top and bottom) have the same ID, happens when I add two new inline-entries in the realtionship-table.